lib/smart_answer/flow.rb

```diff
@@ -147,6 +147,30 @@ def process(responses)
       end
     end
 
+    def resolve_state(responses, requested_node)
+      state = start_state
+
+      until state.nil?
+        return state if state.error
+        return state if state.current_node.to_s == requested_node
```

Comment:
```suggestion
        return state if state.current_node.to_s == requested_node && !responses.key?(state.current_node.to_s)
```
Do we need to check if there's been attempt to respond previously?

Reply:
That looks to work!